keps/sig-node/0035-20190130-topology-manager.md

```diff
@@ -172,39 +173,89 @@ Kubelet components.
 
 If the hints are not compatible, the Topology Manager may choose to
 reject the pod. Behavior in this case depends on a new Kubelet configuration
-value to choose the topology policy. The Topology Manager supports two
-modes: `strict` and `preferred` (default). In `strict` mode, the pod is
-rejected if alignment cannot be satisfied. The Topology Manager could
-use `softAdmitHandler` to keep the pod in `Pending` state.
+value to choose the topology policy. The Topology Manager supports four
+policies: `none`(default), `best-effort`, `restricted` and `single-numa-node`. 
+
+A topology hint indicates a preference for some well-known local resources.
+The Topology Hint currently consists of 
+* A list of bitmasks denoting the possible NUMA Nodes that a request can be satisfied.
+* A preferred field.
+    * This field is defined as follows:
+      * For each Hint Provider, there is a possible resource assignment that satisfies the request, such that the least possible number of NUMA nodes is involed (caculated as if the node were empty.)
+      * There is a possible assignment where the union of involved NUMA nodes for all such resource is no larger than the width required for any single resource
+
+#### Policies
+
+**none (default)**: Kubelet does not consult Topology Manager for placement decisions 
+
+**best-effort**: Topology Manager will provide the preferred allocation for the container based
+on the hints provided by the Hint Providers. If an undesirable allocation is determined, the pod will be admitted with this undesirable allocation.
+
+**restricted**:  Topology Manager will provide the preferred allocation for the container based
+on the hints provided by the Hint Providers. If an undesirable allocation is determined, the pod will be rejected. 
+This will result in the pod being in a `Pending` state.
```

Comment:
This isn't actually true. It doesn't enter the `pending` state. The pod enters the `terminated` state with a pod admission failure. An external controller is required to resubmit the pod at a later time. The scheduler does not attempt to rerun it when new resources are freed.